ToDoApp/ToDoApp/ToDoService.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  ToDoDetail.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-11-29.
+//
+
+import Foundation
+
+enum TaskError: Error {
+    case unavailableTask
+}
+
+struct Task {
+    var id: UUID = UUID()
+    var title: String
+    var note: String?
+    var deadline: Date?
+    var status: Bool = false
+}
+
+protocol DateChecker {
+    func isDateInToday(_ date: Date) -> Bool
+}
+//  conform Calendar to this protocol
+extension Calendar: DateChecker {}
+
+protocol ToDoService {
+    func createTask(title: String, note: String, deadline: Date?) async -> Result<Task, TaskError>
+    func updateTask(todo: Task) async -> Result<Task, TaskError>
+    func taskForToday() async -> [Task]
```

Comment:
let's have this returning a Result as it may also fail

Reply:
This has been fixed.